libmscore/durationtype.cpp

```diff
@@ -304,7 +304,11 @@ void TDuration::shiftType(int nSteps, bool stepDotted)
                   newValue = int(_val) + nSteps;
                   }
 
-            if ((newValue < int(DurationType::V_LONG)) || (newValue > int(DurationType::V_128TH)))
+            if ((newValue < int(DurationType::V_LONG)) || (newValue > int(DurationType::V_128TH)) ||
+                ((newValue == int(DurationType::V_128TH)) && (newDots >= 1)) ||
+                ((newValue == int(DurationType::V_64TH))  && (newDots >= 2)) ||
+                ((newValue == int(DurationType::V_32ND))  && (newDots >= 3)) ||
+                ((newValue == int(DurationType::V_16TH))  && (newDots >= 4)))
```

Comment:
@lasconic If the eventual limits in master is 1024th, then I guess I should make a corresponding master fix which prevents the corresponding cases that would produce 2048th rests...sound good?